WIP: blacklist by removing packages from the set of built and run tests

tylerjw opened the pull request:
To fix the bug where blacklisted packages are removed from the workspace creating a situation where they are linked from debians.

Fixes: https://github.com/ros-planning/moveit/issues/2042

tylerjw committed:
blacklist by removing packages from the set of built and run tests

rhaschke reviewed (commented):
As already pointed out in https://github.com/ros-planning/moveit/issues/2042#issuecomment-622979964, this approach (which we had before) doesn't filter out meta packages.
Building these meta-packages, e.g. ros-perception, essentially pulls in blacklisted packages again.

Your test in https://travis-ci.org/github/ros-planning/moveit/jobs/683524593 is not expressive, because it doesn't declare any blacklisted packages for testing.

tylerjw commented:
By adding the `--no-deps` option to `run_tests` command you won't run the meta-packages (that is my theory).  The travis run is still blacklisting `moveit_ros_perception`.

tylerjw committed:
fix unbound variable errors

rhaschke commented:
> By adding the `--no-deps` option to `run_tests` command you won't run the blacklisted packages because the meta packages won't pull their dependencies (that is my theory).

Maybe. I noticed in the past, that some tests already _run_ with `catkin build tests`.

> The travis run is still blacklisting `moveit_ros_perception`.

Not the one here: https://travis-ci.org/github/ros-planning/moveit/builds/683524588

tylerjw committed:
log blacklist of packages

tylerjw commented:
I think the reason the blacklist isn't getting set is the comment in the `.travis.yml` about it.  I removed that comment to see if that fixes it.

As to the first point, won't calling `run_tests` build the tests?  Should we just change moveit_ci to just call `run_tests` instead of doing it as two steps?

tylerjw committed:
refactor into an if statement

tylerjw commented:
@rhaschke I pushed an update where I refactored the code to be nicer.  I will kick off another build of my PR to validate this.  I also added a comment around the make test and run_tests commands to clarify what we are doing with the whitelist.

rhaschke commented:
@tylerjw, as you can see from the [travis log](https://travis-ci.org/github/ros-planning/moveit_ci/jobs/683703044) `moveit_ros_perception` is run, both by `catkin build tests` and `catkin run_tests`.

tylerjw commented:
@rhaschke I see that.  And yet it doesn't fail the `_preception` test :confused:

tylerjw committed:
try run_tests without explicitly building tests

rhaschke commented:
> And yet it doesn't fail the `_preception` test  (on my moveit PR)

That's because we fixed the perception test on master (https://github.com/ros-planning/moveit/pull/2044). moveit_ci uses the melodic branch for testing, which still has the broken test.

> As you said before, you tried this. Do you have any better ideas on how to fix this (short of fixing the failing test) as it seems `catkin config --blacklist` isn't our friend?

No, sorry. I don't have time to dive into this as well.

tylerjw commented:
> That's because we fixed the perception test on master ([ros-planning/moveit#2044](https://github.com/ros-planning/moveit/pull/2044)). moveit_ci uses the melodic branch for testing, which still has the broken test.

Wow, that's awesome, I didn't see that.  In that case I'll be able to continue on debugging my test by just rebasing on master without having to fix this too.

tylerjw commented:
Setting this as WIP as it doesn't solve the problem and I don't have any better ideas yet.

tylerjw commented:
Closing as this is no longer needed or useful.
